lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -1280,6 +1280,112 @@ void BoundInterpreterFunction::fwdBatchOneHotInst(
   }
 }
 
+// Example:
+// Depth 0                    Depth 1
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+//
+// after transformation
+// blockDepth0
+//  depth 0          depth 1
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// blockDepth 1
+//  depth 2          depth 3
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// blockDepth 2
+// depth 4           depth 5
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// blockDepth 3
+// depth 6           depth 7
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+template <typename ElemTy>
+void BoundInterpreterFunction::fwdSpaceToDepthInstImpl(
+    const glow::SpaceToDepthInst *I) {
+  auto *inT = getTensor(I->getSrc());
+  auto *outT = getTensor(I->getDest());
+
+  ElemTy *inPtr = reinterpret_cast<ElemTy *>(inT->getUnsafePtr());
+  ElemTy *outPtr = reinterpret_cast<ElemTy *>(outT->getUnsafePtr());
+
+  unsigned blockSize = I->getBlockSize();
+
+  size_t inHeight = inT->dims()[1];
+  size_t inWidth = inT->dims()[2];
+  size_t inDepth = inT->dims()[3];
+
+  size_t outBatch = outT->dims()[0];
+  size_t outHeight = outT->dims()[1];
+  size_t outWidth = outT->dims()[2];
+  size_t outDepth = outT->dims()[3];
+
+  for (size_t b = 0; b < outBatch; ++b) {
+    for (size_t h = 0; h < outHeight; ++h) {
```

Comment:
Let's rename `h` to `oh`, and so on. So that it's clear that we are iterating over output.